rpc/kate-rpc/src/lib.rs

```diff
@@ -409,16 +448,10 @@ where
 			.ok_or_else(|| internal_err!("Missing block hash {:?}", at))?
 			.block;
 
-		// Get Opaque Extrinsics and transform into AppUncheckedExt.
-		let calls = block
-			.extrinsics()
-			.iter()
-			.filter_map(|opaque| UncheckedExtrinsic::try_from(opaque).ok())
```

Comment:
This seemed to suffer from the same problem, except here, we need calls to be decoded, app_id is not enough. The decoded extrinsics are then sent to runtime to filter out just Avail Submit Data extrinsics, and the proof is then created (also inside runtime). Upon closer inspection, there is already a public function called extrinsics_proof that instead does the decoding also in runtime, which solves the problem. However, that was probably used to save memory, because in this instance, all the extrinsics are sent over to runtime, instead of just Avail data. There's a todo there about exact length iterator requirements, which makes me think that.

Reply:
would be good to se what are benchmark performance on `extrinsics_proof` vs `calls_proof`